java/src/main/java/ai/rapids/cudf/DecimalUtils.java

```diff
@@ -142,7 +144,16 @@ public static ColumnVector greaterThan(ColumnView lhs, BigDecimal rhs) {
     // For example:
     //      100.2 > 100.19
     // If we rounded up the rhs 100.19 would become 100.2, and now 100.2 is not > 100.2
-    BigDecimal roundedRhs = rhs.setScale(-cvScale, BigDecimal.ROUND_DOWN);
+
+    // First we have to round the scalar (rhs) to the same scale as lhs.
+    // Because this is a greater-than, and it is rhs that we are rounding, we will round towards 0 (DOWN) in case rhs is
+    // positive and away from 0 (UP) if rhs is negative to make sure we always return the correct value.
+    // For example:
+    //      ex:1 100.2 > 100.19
+    //      ex:2 -100.1 > -100.19
+    // In ex:1 If we rounded up the rhs 100.19 would become 100.2, and now 100.2 is not > 100.2
+    // In ex:2 If we rounded down the rhs -100.19 would become -100.1, and now -100.1 is not > -100.1
+    BigDecimal roundedRhs = rhs.setScale(-cvScale, rhs.signum() > 0 ? BigDecimal.ROUND_DOWN : BigDecimal.ROUND_UP);
```

Comment:
This is just repeated of the code above. We should not copy-paste code around. If there is the same code used multiple times, let's extract it into a function and call it multiple times instead.